test-utils/src/main/java/org/dmfs/testutils/tools/AbstractValueObject.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.testutils.tools;
+
+/**
+ * Base class for test value object classes.
+ *
+ * @author Gabor Keszthelyi
+ */
+public abstract class AbstractValueObject
```

Comment:
Still not a big fan of this. I've created https://github.com/mockito/mockito/issues/1205 because "equal mocks" look like a better solution to me.
If you don't need to mock a specific class or interface you could also just use `new String("1")` which will always create a new instance which equals any other `new String("1")`. If you need an object of another type `new Integer(1)` might do the job.

Reply:
Ok, I've removed it. 
True that we can certainly use `new String()` and `new Integer()`, this would have been only for making that intent explicit, to use equal but different instances. (`new String()` and `new Integer()` gives warnings now to use the literals)